scripts/quests/outlands/The_Search_for_Goldmane.lua

```diff
@@ -0,0 +1,259 @@
+                end,
+            },
+
+            ['Trunk'] =
+            {
+                onTrigger = function(player, npc)
+                    if quest:getVar(player, 'Prog') == 2 then
+                        return quest:messageSpecial(riverneID.text.TRUNK_EXAMINE)
+                    elseif quest:getMustZone(player) then
+                        return quest:messageSpecial(riverneID.text.TRUNK_QUEST_INVESTIGATED)
+                    elseif quest:getVar(player, 'Prog') >= 3 then
+                        return quest:messageSpecial(riverneID.text.TRUNK_INVESTIGATION_COMPLETE)
+                    end
+                end,
+
+                onTrade = function(player, npc, trade)
+                    if
+                        quest:getVar(player, 'Prog') == 2 and
+                            npcUtil.tradeHasExactly(trade, xi.item.COPPER_KEY)
+                    then
+                        return quest:progressEvent(41)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [40] = function(player, csid, option, npc)
+                    if option == 1 then
+                        quest:setVar(player, 'Prog', 2)
+                    end
+                end,
+
+                [41] = function(player, csid, option, npc)
+                    quest:setVar(player, 'Prog', 3)
+                    player:tradeComplete()
+                    quest:setMustZone(player)
+                end,
+            },
+        },
+    },
+
+    {
+        -- Quest stage in Bastok Purgonorgo Isle
+        check = function(player, status, vars)
+            return status == xi.questStatus.QUEST_ACCEPTED and vars.Prog >= 3
+        end,
+
+        [xi.zone.METALWORKS] =
+        {
+            ['Vladinek'] =
+            {
+                onTrigger = function(player, npc)
+                    if quest:getVar(player, 'Prog') == 3 then
+                        return quest:progressEvent(887)
+                    elseif quest:getVar(player, 'Prog') == 4 then
+                        return quest:event(888)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [887] = function(player, csid, option, npc)
+                    quest:setVar(player, 'Prog', 4)
+                end,
+            },
+        },
+
+        [xi.zone.BIBIKI_BAY] =
+        {
+            ['Weathered_Boat'] =
+            {
+                onTrigger = function(player, npc)
+                    if os.time() >= npc:getLocalVar('Wait') then
+                        if quest:getVar(player, 'Prog') == 4 then
+                            return quest:progressEvent(40)
+                        elseif quest:getVar(player, 'Prog') == 5 then
+                            npc:setLocalVar('QuestPlayer', player:getID())
+                            npcUtil.popFromQM(player, npc, bibikiID.mob.ROHEMOLIPAUD_OFFSET, { claim = true, hide = 0 })
```

Comment:
if pop from npc then
    npc:setLocalVar()
    return message
end
so it doesnt set the npc var unless they pop